src/Sarif.Multitool/Rules/SARIF1008.PhysicalLocationPropertiesMustBeConsistent.cs

```diff
@@ -18,15 +21,130 @@ public class PhysicalLocationPropertiesMustBeConsistent : SarifValidationSkimmer
+                return;
+            }
+
+            if (!IsRegionProperSuperset(physicalLocation.ContextRegion, physicalLocation.Region))
+            {
+                LogResult(physicalLocationPointer, nameof(RuleResources.SARIF1008_PhysicalLocationPropertiesMustBeConsistent_Error_ContextRegionMustBeProperSupersetOfRegion_Text));
+            }
+        }
+
+        private static bool IsRegionProperSuperset(Region superRegion, Region subRegion)
+        {
+            if (IsLineColumnBasedTextRegion(superRegion) &&
+                IsLineColumnBasedTextRegion(subRegion) &&
+                !IsLineColumnBasedTextRegionProperSuperset(superRegion, subRegion))
+            {
+                return false;
+            }
+
+            if (IsOffsetBasedTextRegion(superRegion) &&
+                IsOffsetBasedTextRegion(subRegion) &&
+                !IsOffsetBasedTextRegionProperSupetSet(superRegion, subRegion))
+            {
+
+                return false;
             }
+
+            if (IsBinaryRegion(superRegion) &&
+                IsBinaryRegion(subRegion) &&
+                !IsBinaryRegionProperSuperset(superRegion, subRegion))
+            {
+                return false;
+            }
+
+            // if we reach here, the region and context region have been expressed as different property sets,
+            // and it is not possible to judge validity without looking at the actual content.
+            // It is a potential false negative.
+            return true;
+
+        }
+
+        private static bool IsBinaryRegionProperSuperset(Region superRegion, Region subRegion)
+        {
+            if (superRegion.ByteOffset > subRegion.ByteOffset)
+            {
+                return false;
+            }
+
+            if (superRegion.ByteOffset == subRegion.ByteOffset && superRegion.ByteLength <= subRegion.ByteLength)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool IsLineColumnBasedTextRegionProperSuperset(Region superRegion, Region subRegion)
+        {
+            if (superRegion.StartLine > subRegion.StartLine || superRegion.EndLine < subRegion.EndLine)
+            {
+                return false;
+            }
+
+            if (superRegion.StartLine == subRegion.StartLine && superRegion.StartColumn < subRegion.StartColumn)
+            {
+                return false;
+            }
+
+            if (superRegion.EndLine == subRegion.EndLine && superRegion.EndColumn > subRegion.EndColumn)
+            {
+                return false;
+            }
+
+            if (superRegion.StartLine == subRegion.StartLine &&
+                superRegion.EndLine == subRegion.EndLine &&
+                superRegion.StartColumn == subRegion.StartColumn &&
+                superRegion.EndColumn == subRegion.EndColumn)
+            {
```

Comment:
all these could be a single if condition with a bunch of "ors". i went back an forth on it and ultimately decided, this form is better for quicker undersantding of all the things we check. i follow same pattern in parent method as well.

Reply:
I'm fine with splitting it. A seven-clause OR statement could be a bit much.

---
In reply to: [443859402](https://github.com/microsoft/sarif-sdk/pull/1925#discussion_r443859402) [](ancestors = 443859402)